### ISSUE TITLE
STK: Do not use deprecated `KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_*` macros

### ISSUE THREAD

**Issue by dalg24** [trilinos/Trilinos#11509]
## Bug Report
@alanw0
@ndellingwood 

Compatibility with Kokkos

### Description
See kokkos/kokkos#4668 or (better) the [release 3.6 briefing slides](https://github.com/kokkos/kokkos-tutorials/blob/main/Other/ReleaseBriefings/release-36.pdf) p45-50

Please make sure you read the warning on the last page.  Don't just go and replace with `KOKKOS_IF_ON_{HOST,DEVICE}`, consider whether you can deduce the execution space and specialize or use `if constexpr`.

These are not portable and we are about to remove them.

```
packages/stk/stk_expreval/stk_expreval/DeviceVariableMap.hpp:60:#ifndef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
packages/stk/stk_expreval/stk_expreval/Function.hpp:159:#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
packages/stk/stk_expreval/stk_expreval/Function.hpp:172:#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
packages/stk/stk_expreval/stk_expreval/Function.hpp:185:#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
packages/stk/stk_expreval/stk_expreval/Function.hpp:204:#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
packages/stk/stk_expreval/stk_expreval/Function.hpp:228:#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
packages/stk/stk_expreval/stk_expreval/Function.hpp:244:#if defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
packages/stk/stk_expreval/stk_expreval/ParsedEval.hpp:100:#ifndef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
packages/stk/stk_expreval/stk_expreval/ParsedEval.hpp:109:#ifndef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
packages/stk/stk_mesh/stk_mesh/base/DeviceMesh.hpp:463:    #ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
packages/stk/stk_ngp_test/stk_ngp_test/GlobalReporter.cpp:27:  #ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HIP_GPU

**Comment by ndellingwood** [comment]
Adding @ldh4 as well

**Comment by dalg24** [comment]
Will be fixed by #11599
